comp/core/configsync/configsyncimpl/module.go

```diff
@@ -63,36 +63,41 @@ type configSync struct {
 
 // newComponent checks if the component was enabled as per the config and return a enable/disabled configsync
 func newComponent(deps dependencies) (configsync.Component, error) {
-	agentIPCPort := deps.Config.GetInt("agent_ipc.port")
-	configRefreshIntervalSec := deps.Config.GetInt("agent_ipc.config_refresh_interval")
+	var compURL *url.URL
+	if deps.Config.GetBool("agent_ipc.use_socket") {
+		path := deps.Config.GetString("agent_ipc.socket_path") + "/agent_ipc.socket"
+		compURL = &url.URL{
+			Scheme: "http",
```

Comment:
Should we not use HTTPS for the socket too ?

Reply:
fixed in https://github.com/DataDog/datadog-agent/pull/43180/commits/bfffaad8d070e57d59429d1ec1dbcd53e4984fe7